src/commands/Twitch/twitchsubscription.ts

```diff
@@ -188,18 +186,33 @@ export default class extends SkyraCommand {
 
 		// If the streamer was not defined, reset all entries and purge all entries.
 		if (typeof streamer === 'undefined') {
-			const entries = subscriptions.reduce((accumulator, subscription) => accumulator + subscription[1].length, 0);
-			if (entries === 0) throw message.language.tget('COMMAND_TWITCHSUBSCRIPTION_RESET_EMPTY');
+			const subscription = subscriptions.reduce((accumulator, subscription) => accumulator + subscription[1].length, 0);
+			if (subscription === 0) throw message.language.tget('COMMAND_TWITCHSUBSCRIPTION_RESET_EMPTY');
 			await message.guild!.settings.reset($KEY);
 
 			// Update all entries that include this guild, then iterate over the empty values and remove the empty ones.
-			const updated = await this.client.queries.purgeTwitchStreamGuildSubscriptions(message.guild!.id);
-			const pendingToDelete: string[] = [];
-			for (const update of updated) {
-				if (update.guild_ids.length === 0) pendingToDelete.push(update.id);
-			}
-			if (updated.length !== 0) await this.client.queries.deleteTwitchStreamSubscriptions(pendingToDelete);
-			return message.sendLocale('COMMAND_TWITCHSUBSCRIPTION_RESET_SUCCESS', [entries]);
+			const connection = await DbSet.connect();
+			await connection.twitchStreamSubscriptions.manager.transaction(async em => {
+				const entries = await em.find(TwitchStreamSubscriptionEntity, { guildIds: Any([message.guild!.id]) });
+				const toUpdate: TwitchStreamSubscriptionEntity[] = [];
+				const toDelete: TwitchStreamSubscriptionEntity[] = [];
+				for (const entry of entries) {
+					if (entry.guildIds.length === 1) {
```

Comment:
It was `=== 0` before?

Reply:
Because the entry was removed if it became empty, we optimized this flow by going from:

`remove from array` -> `check if array is empty` -> `delete from database`

to:

`get all entries that contain the guild` -> `check if the array will be empty` -> `update or delete from database`